### PR TITLE
Adding pretender gem

### PR DESCRIPTION
**Pull request by matsevich**
[Trello Task](https://trello.com/c/BtV1rKTg/61-admin-user-impersonate)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

Screenshot from 2020-11-09 12-07-36

Screenshot from 2020-11-08 02-41-51
Screenshot from 2020-11-08 02-42-16

### CODE REVIEW

**Review by Axedez — APPROVED** [review]

File: app/controllers/account/admin/admin_controller.rb [hunk 1]
```diff
@@ -1,2 +1,12 @@
 class Account::Admin::AdminController < Account::AccountController
+  def start_impersonate
+    user = User.find_by(id: params[:user_id])
+    impersonate_user(user) if user.present?
```
Comment (mmarusyk) [inline app/controllers/account/admin/admin_controller.rb:4]: Enter

**Review by mmarusyk — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class ApplicationController < ActionController::Base
+  impersonates :user
   before_action :configure_permitted_parameters, if: :devise_controller?
   layout :choose_layout
```
Comment (BogdanBusko) [inline app/controllers/application_controller.rb:4]: it must be at the top

File: config/routes.rb [hunk 3]
```diff
@@ -33,6 +33,8 @@
         post 'import', on: :collection
       end
       resources :users, only: %i[index destroy]
+      get 'start_impersonate' => 'admin#start_impersonate', as: 'start_impersonate'
```
Comment (BogdanBusko) [inline config/routes.rb:36]: get 'start_impersonate', to: .....

File: app/controllers/account/admin/admin_controller.rb [hunk 4]
```diff
@@ -1,2 +1,13 @@
 class Account::Admin::AdminController < Account::AccountController
+  def start_impersonate
+    user = User.find_by(id: params[:user_id])
```
Comment (chu4) [inline app/controllers/account/admin/admin_controller.rb:3]: there should not be a possibility to impersonate another admin

File: app/views/account/admin/users/_user.html.slim [hunk 5]
```diff
@@ -14,4 +14,6 @@
 
     div
       = link_to t('.btn-show'), [:account, user], class: 'btn btn-outline-primary'
+      = link_to 'Impersonate', account_admin_start_impersonate_path(user_id: user.id),
```
Comment (chu4) [inline app/views/account/admin/users/_user.html.slim:17]: no button if user_to_impersonate is an admin

File: app/controllers/account/admin/admin_controller.rb [hunk 6]
```diff
@@ -1,2 +1,13 @@
 class Account::Admin::AdminController < Account::AccountController
+  def start_impersonate
+    user = User.where("id = ? AND role != ?", params[:user_id], 0).take
```
Comment (chu4) [inline app/controllers/account/admin/admin_controller.rb:3]: create a scope non_admin in User model and just apply it here before searcing by id
don't use integers of roles 0 you specifically have :admin symbol for it

File: app/controllers/account/admin/admin_controller.rb [hunk 7]
```diff
@@ -1,2 +1,13 @@
...
+    impersonate_user(user) if user.present?
+
```
Comment (chu4) [inline app/controllers/account/admin/admin_controller.rb:5]: add error message "You can't impersonate this user" if there is no user
don't redirect on error

**Review by chu4 — APPROVED** [review]

**Pull request merged**